src/pnpq/devices/odl_ozoptics_650ml.py

```diff
@@ -4,63 +4,135 @@
 import serial
 from serial import Serial
 from pnpq.devices.optical_delay_line import OpticalDelayLine
-import time
+import time, logging
+from pnpq.errors import (
+    DeviceDisconnectedError,
+    OdlMoveNotCompleted,
+    OdlHomeNotCompleted,
+    OdlGetPosNotCompleted,
+    OdlMoveOutofRangeError,
+)
+
 
 
 class OdlOzOptics(OpticalDelayLine):
+    max_move: int
+    """maximum length for moving odl from 0-position, the total path length will be 2x(max_move)"""
+
+    min_move: int
+    """minimum specified length for moving: usually zero"""
+
+    resolution: int
+    """represents minimum specified length for moving on the stage: usually zero"""
+
+    home_timeout: int
+    """required time for completing home in seconds"""
+
+    move_timeout: int
+    """required time for completing home in seconds"""
+
     def __init__(
         self,
         serial_port: str | None = None,
         serial_number: str | None = None,
         config_file=None,
     ):
         super().__init__(serial_port, serial_number, config_file)
+        self.name = "OzOptics ODL"
+
         self.conn.baudrate = 9600
         """Basic Communication BaudRate"""
         self.resolution = 32768 / 5.08
         """32768 steps per motor revolution(5.08 mm = 2xDistance Travel or mirror travel per pitch 0.1 inch)"""
-
+        self.logger = logging.getLogger(f"{self}")
         self.command_terminate = "\r\n"
+        self.max_move = 50
+        self.min_move = 0
+        self.logger = logging.getLogger(f"{self}")
 
         try:
             self.conn.open()
+
         except:
             raise RuntimeError("Can not open OZ optic ODL device")
 
-    def connect(self):
-        if self.conn.is_open == 0:
+    def __ensure_port_open(self) -> None:
+        if not self.conn.is_open:
+            self.logger.error("disconnected")
+            raise DeviceDisconnectedError(f"{self} is disconnected")
+
+    def reconnect(self):
+        if self.conn.is_open:
+            self.logger.warning(f"Serial connectio to ODL({self}) is already open!")
+        else:
+            self.logger.info(f"Openning connection to ODL({self}) device!")
             try:
                 self.conn.open()
-            except Exception as err:
-                raise RuntimeError("Connection failed: " + str(err))
+            except:
+                raise DeviceDisconnectedError(f"{self} is disconnected")
```

Comment:
This error is not an accurate description of what is happening.

The old code was more accurate, but it is probably still not correct.

The best thing to do here might be to remove the try/except logic completely. Just let any exceptions from `self.conn.open()` continue up the stack.